llm-translator.html

```diff
@@ -0,0 +1,191 @@
+            call.updateParticipant(p["session_id"], {
+              setSubscribedTracks: {
+                video: true,
+                audio: audioSub,
+              },
+            });
+          }
+        });
+
+        call.on("app-message", (e) => {
+          console.log("Got app message: ", e);
+          const lang = languageSelect.value;
+          if (e.fromId === "transcription" && lang === "english") {
+            console.log("got it for english: ", e.data.text);
+            showSubtitle(e.data.session_id, e.data.text);
+          } else if (e.data.translation_language === lang) {
+            console.log("got it for non-english: ", e.data.translation);
+            showSubtitle(e.data.session_id, e.data.translation);
+          }
+          e.fromId;
+        });
+      }
+
+      function startTrack(evt) {
+        console.log("track started: ", evt);
+        if (evt.track.kind === "audio" && !evt.participant.local) {
+          let audiosDiv = document.getElementById("audios");
+          let audioEl = document.createElement("audio");
+          audiosDiv.appendChild(audioEl);
+          audioEl.style.width = "100%";
+          audioEl.srcObject = new MediaStream([evt.track]);
+          audioEl.play();
+        } else if (evt.track.kind === "video") {
+          let pid = evt.participant.session_id;
+          console.log({ pid });
+          let videosDiv = document.getElementById("videos");
+          let paxDiv = document.createElement("div");
+          paxDiv.id = "pax-" + pid;
+          let videoEl = document.createElement("video");
+          let subtitleDiv = document.createElement("div");
+          subtitleDiv.id = "subtitles-" + pid;
+          paxDiv.appendChild(videoEl);
+          paxDiv.appendChild(subtitleDiv);
+          videosDiv.appendChild(paxDiv);
+          paxDiv.style.width = "50%";
+          videoEl.style.width = "100%";
+          videoEl.srcObject = new MediaStream([evt.track]);
+          videoEl.play();
+        }
+      }
+
+      function stopTrack(evt) {
+        let vids = document.getElementsByTagName("video");
+        for (let vid of vids) {
+          if (
+            vid.srcObject &&
+            vid.srcObject.getVideoTracks()[0] === evt.track
+          ) {
+            vid.remove();
+          }
+        }
+      }
+
+      async function changeLanguage() {
+        const lang = languageSelect.value;
+        console.log("changing language to: ", lang);
+        let allPax = call.participants();
+        let paxIds = Object.keys(allPax).filter((p) => p !== "local");
+        paxIds.forEach((id) => {
+          const p = allPax[id];
+          const wantsAudio =
+            (isTranslator(p) && translatorLanguage(p) === lang) ||
+            (!isTranslator(p) && lang === "english");
+          call.updateParticipant(id, {
+            setSubscribedTracks: { audio: wantsAudio },
+          });
+        });
+      }
+
+      async function joinRoom() {
+        await call.join({});
```

Comment:
I wonder if we need this separate function, if all it does is one thing maybe it's more readable to keep in in-line?